examples/experimental/weather/training_recipe/train.py

```
@@ -0,0 +1,241 @@
+        ],  # TODO (mnabian): check this
+        batch_size=cfg.training.batch_size,
+        stride=cfg.stride,
+        dt=cfg.dt,
+        start_year=cfg.training.start_year,
+        num_steps=cfg.training.num_steps,
+        lsm_filename=to_absolute_path(cfg.lsm_filename),
+        geopotential_filename=to_absolute_path(cfg.geopotential_filename),
+        use_latlon=cfg.use_latlon,
+        use_cos_zenith=cfg.use_cos_zenith,
+        patch_size=cfg.patch_size,
+        num_samples_per_year=cfg.training.num_samples_per_year,
+        shuffle=True,
+        num_workers=cfg.num_workers,
+        device=dist.device,
+        process_rank=dist.rank,
+        world_size=dist.world_size,
+    )
+    logger.success(f"Loaded datapipe of size {len(datapipe)}")
+
+    # instantiate the validation datapipe
+    if dist.rank == 0:
+        logger.file_logging()
+        validation_datapipe = ClimateHDF5Datapipe(
+            data_dir=cfg.validation.data_dir,
+            # stats_dir=cfg.stats_dir,   # TODO (mnabian): uncomment
+            channels=[
+                i for i in range(cfg.model.in_channels - cfg.num_static_channels)
+            ],
+            batch_size=cfg.validation.batch_size,
+            stride=cfg.stride,
+            dt=cfg.dt,
+            start_year=cfg.validation.start_year,
+            num_steps=cfg.validation.num_steps,
+            lsm_filename=to_absolute_path(cfg.lsm_filename),
+            geopotential_filename=to_absolute_path(cfg.geopotential_filename),
+            use_latlon=cfg.use_latlon,
+            use_cos_zenith=cfg.use_cos_zenith,
+            patch_size=cfg.patch_size,
+            num_samples_per_year=cfg.validation.num_samples_per_year,
+            num_workers=cfg.num_workers,
+            shuffle=False,
+            device=dist.device,
+        )
+        logger.success(f"Loaded validation datapipe of size {len(validation_datapipe)}")
+
+    # instantiate the model
+    registry = ModelRegistry()
+    assert (
```

The `Module.instantiate` call should raise an exception with the same message if the model is not found. Should let the check happen inside that method in case the implementation there changes.

---

Fixed this